src/framework/components/anim/component.js

```diff
@@ -447,16 +447,15 @@ class AnimComponent extends Component {
         this._layerIndices = {};
         this._parameters = {};
         this._playing = false;
-        Object.keys(this._targets).forEach((targetKey) => {
-            this._targets[targetKey].unbind();
-        });
+        this.unbind();
         // clear all targets from previous binding
         this._targets = {};
     }
 
     resetStateGraph() {
+        const assetId = this._stateGraphAsset;
         this.removeStateGraph();
-        if (this.stateGraphAsset) {
+        if (assetId) {
             const stateGraph = this.system.app.assets.get(this.stateGraphAsset).resource;
```

Comment:
```suggestion
            const stateGraph = this.system.app.assets.get(assetId).resource;
```
Should this be `assetId` instead?